### PR TITLE
Add unused_setter_argument rule

### LINKED ISSUE #1136: Rule request: newValue ignored on var set
I've seen a lot of times people using the variable name instead of `newValue` which leads to bugs:

```swift
var aValue: String {
    get {
        return Persister.shared.aValue
    }
    set {
        Perister.shared.aValue = aValue //should be newValue
    }
}
```

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #1136

~Still WIP because it's not handling explicitly named arguments. Just wanted to see oss-check.~

### CODE REVIEW

**Comment by jpsim** [comment]
How about `unused_setter_value` instead?

**Comment by marcelofabri** [comment]
I don't have a strong preference, will rename it 👍

File: Source/SwiftLintFramework/Rules/Lint/UnusedSetterArgumentRule.swift [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        description: "Setter argument is not used.",
+        kind: .lint,
+        nonTriggeringExamples: [
+            """
+            var aValue: String {
+                get {
+                    return Persister.shared.aValue
+                }
+                set {
+                    Perister.shared.aValue = newValue
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/UnusedSetterArgumentRule.swift]: Nit: `Perister` here and a few other places

Comment (marcelofabri) [reply Source/SwiftLintFramework/Rules/Lint/UnusedSetterArgumentRule.swift]: Copied from #1136 😅

**Pull request merged**